GoogleForCreators/web-stories-wp
# Dashboard: Load Spinner and Failure to load resource

## Context


Fix issues stories not loading on 'infinite scroll'

## Summary


The Dashboards infinite scroll feature is heavily reliant on `useStoryView` to adjust the `page` query param while fetching stories.  The infinite scroll increases the `page` query param and we need to reset that when filtering and sorting.

## Relevant Technical Choices


Adds FiltersProvider integration with the Dashboard infinite scroll functionality.
## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Go to the Dashboard
2. Make sure you have enough stories to trigger the 'infinite scroll' functionality. I believe we fetch 24 stories at a time so you'd need at least 25
- Notice the main issue stated in this ticket - the loading spinner not being removed, is now fixed due to #11736 
3. Once the next set of stories is fetched, change one of the filters that has less than 2 pages worth of stories (so less than 25 stories)

On Staging:
- Notice a 'snackbar' popup 'Unable to load stories' and you should have received a `400 Failed to load resource:` error in you console / network tab.

On Branch:
- Notice no 'snackbar' popup, no `400` error, and filters should work as expected.



## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11716
